test/e2e/simple_preemptor/simple_preemptor_test.go

```diff
@@ -0,0 +1,182 @@
+		sleepPod3Configs := k8s.SleepPodConfig{Name: "sleepjob3", NS: dev, Mem: 2000, Time: 600, RequiredNode: "yk8s-worker"}
+
+		// random uid in pod spec to ensure this Non-opted pod doesn't become originator
+		sleepPod4Configs := k8s.SleepPodConfig{Name: "sleepjob4", NS: dev, Mem: 2500, Time: 600, Optedout: true, UID: types.UID("ddd")}
+		sleepPod5Configs := k8s.SleepPodConfig{Name: "sleepjob5", NS: dev, Mem: 2500, Time: 600, Optedout: false}
+		sleepPod6Configs := k8s.SleepPodConfig{Name: "sleepjob6", NS: dev, Mem: 2500, Time: 600, Optedout: false}
+		sleepPod7Configs := k8s.SleepPodConfig{Name: "sleepjob7", NS: dev, Mem: 2500, Time: 600, RequiredNode: "yk8s-worker2"}
+
+		for _, config := range []k8s.SleepPodConfig{sleepPodConfigs, sleepPod2Configs,
+			sleepPod3Configs, sleepPod4Configs, sleepPod5Configs, sleepPod6Configs} {
+			ginkgo.By("Deploy the sleep pod " + config.Name + " to the development namespace")
+			sleepObj, podErr := k8s.InitSleepPod(config)
+
+			Ω(podErr).NotTo(gomega.HaveOccurred())
+			sleepRespPod, err := kClient.CreatePod(sleepObj, dev)
+			gomega.Ω(err).NotTo(gomega.HaveOccurred())
+			// Wait for pod to move to running state
+			err = kClient.WaitForPodBySelectorRunning(dev,
+				fmt.Sprintf("app=%s", sleepRespPod.ObjectMeta.Labels["app"]),
+				240)
+			gomega.Ω(err).NotTo(gomega.HaveOccurred())
+		}
+
+		ginkgo.By("Deploy the sleep pod " + sleepPod7Configs.Name + " to the development namespace")
+		sleepObj, podErr := k8s.InitSleepPod(sleepPod7Configs)
+
+		Ω(podErr).NotTo(gomega.HaveOccurred())
+		sleepRespPod, err := kClient.CreatePod(sleepObj, dev)
+		gomega.Ω(err).NotTo(gomega.HaveOccurred())
+
+		// Wait for pod to move to running state
+		err = kClient.WaitForPodBySelectorRunning(dev,
+			fmt.Sprintf("app=%s", sleepRespPod.ObjectMeta.Labels["app"]),
+			600)
+		gomega.Ω(err).NotTo(gomega.HaveOccurred())
+
+		// assert sleeppod6 is killed
+		err = kClient.WaitForPodEvent(dev, "sleepjob4", "Killing", 1200)
+		gomega.Ω(err).NotTo(gomega.HaveOccurred())
+
+	})
+
+	ginkgo.AfterEach(func() {
+
+		// Delete all sleep pods
+		ginkgo.By("Delete all sleep pods")
+		pods, err := kClient.GetPodNamesFromNS(ns.Name)
+		if err == nil {
+			for _, each := range pods {
+				if strings.Contains(each, "sleep") {
+					ginkgo.By("Deleting sleep pod: " + each)
+					err = kClient.DeletePod(each, ns.Name)
+					if err != nil {
+						if statusErr, ok := err.(*k8serrors.StatusError); ok {
+							if statusErr.ErrStatus.Reason == metav1.StatusReasonNotFound {
+								fmt.Fprintf(ginkgo.GinkgoWriter, "Failed to delete pod %s - reason is %s, it "+
+									"has been deleted in the meantime\n", each, statusErr.ErrStatus.Reason)
+								continue
+							}
+						}
+					}
+				}
+			}
+		} else {
+			fmt.Fprintf(ginkgo.GinkgoWriter, "Failed to get pods from namespace %s - reason is %s\n", ns.Name, err.Error())
+		}
+	})
+
+	ginkgo.AfterSuite(func() {
```

Comment:
Add the part which restores the old config map.

Eg. this is from the restart and recovery test:
```
		ginkgo.By("Restoring the old config maps")
		var c, err1 = kClient.GetConfigMaps(configmanager.YuniKornTestConfig.YkNamespace,
			configmanager.DefaultYuniKornConfigMap)
		Ω(err1).NotTo(gomega.HaveOccurred())
		Ω(c).NotTo(gomega.BeNil())
		c.Data = oldConfigMap.Data
		var e, err3 = kClient.UpdateConfigMap(c, configmanager.YuniKornTestConfig.YkNamespace)
		Ω(err3).NotTo(gomega.HaveOccurred())
		Ω(e).NotTo(gomega.BeNil())
```